### PR TITLE
Release/v0.4.3

### PR DESCRIPTION
**Pull request by johnkrovitch**
Improvements :
- Improve request filtering process
- Improve testing
- Separate DataProviders creation from Admin class

Bugfix : 
- Fix error on translation key (#63)
- Fix display of create notification (#58)
- Fix order parameter for action (#59)

### CODE REVIEW

File: Controller/CRUDController.php [hunk 1]
```diff
@@ -121,7 +125,7 @@ public function batchAction(Request $request)
      *
      * @Template("LAGAdminBundle:CRUD:edit.html.twig")
```
Comment (lutangar) [inline Controller/CRUDController.php]: template annotation might not be great for extendability

File: Filter/PagerfantaFilter.php [hunk 2]
```diff
@@ -3,37 +3,37 @@
 namespace LAG\AdminBundle\Filter;
 
 use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\HttpFoundation\ParameterBag;
 
-class PagerfantaFilter
+class PagerfantaFilter extends RequestFilter
```
Comment (lutangar) [inline Filter/PagerfantaFilter.php]: Maybe only the adapter should hold the logic of how PagerFanta is filtering the request.

File: Resources/views/CRUD/list.html.twig [hunk 3]
```diff
@@ -31,66 +31,71 @@
...
-                                            {#<a href="{{ appendToQueryString(app.request, 'export', export) }}">{{ export }}</a>#}
-                                        {#{% endfor %}#}
-                                    {#</td>#}
-                                {#</tr>#}
-                                {#</tfoot>#}
-                            {#{% endif %}#}
-                        </table>
+                        {% endfor %}
+                        </tbody>
+                        {#{% if admin.currentAction.configuration.exports | length %}#}
```
Comment (lutangar) [inline Resources/views/CRUD/list.html.twig]: comments leftovers

File: Tests/Utils/FakeEntityManager.php [hunk 4]
```diff
@@ -0,0 +1,536 @@
...
+use Doctrine\ORM\PessimisticLockException;
+use Doctrine\ORM\Query;
+use Doctrine\ORM\Query\ResultSetMapping;
+use Doctrine\ORM\QueryBuilder;
+use Doctrine\ORM\UnitOfWork;
+
+/**
+ *
+ */
+class FakeEntityManager implements EntityManagerInterface
```
Comment (lutangar) [inline Tests/Utils/FakeEntityManager.php]: Maybe a mock was enough.

File: Controller/CRUDController.php [hunk 5]
```diff
@@ -24,8 +23,6 @@
  */
 class CRUDController extends Controller
```
Comment (johnkrovitch) [inline Controller/CRUDController.php]: #80 created

File: Tests/Utils/FakeEntityManager.php (hunk 4 above)
Comment (johnkrovitch) [inline Tests/Utils/FakeEntityManager.php]: It is more convenient to have a fake class when the container builder compile the service in the tests than a mock. I try to compile the builder to test if the all the services are correctly set.

File: Admin/Admin.php [hunk 6]
```diff
@@ -297,51 +304,67 @@ public function generateRouteName($actionName)
...
+        if (!$currentAction->isLoadingRequired()) {
+            return;
+        }
+        $pager = $currentActionConfiguration->getParameter('pager');
+
+        if ($currentAction->isPaginationRequired() && $pager) {
+            $loadStrategy = $currentActionConfiguration->getParameter('load_strategy');
+
+            // only pagerfanta adapter is yet supported
+            if ('pagerfanta' !== $pager) {
```
Comment (johnkrovitch) [inline Admin/Admin.php]: #82 has been created

File: Admin/Admin.php [hunk 7]
```diff
@@ -519,4 +544,54 @@ protected function generateMessageTranslationKey($message)
...
+     * @param int $offset
+     *
+     * @return array|\Traversable
+     */
+    protected function loadPaginate(array $criteria, array $orderBy, $limit, $offset)
+    {
+        // adapter to pagerfanta
+        $adapter = new PagerFantaAdminAdapter($this->dataProvider, $criteria, $orderBy);
+        // create pager
+        $this->pager = new Pagerfanta($adapter);
```
Comment (johnkrovitch) [inline Admin/Admin.php]: #82

File: Admin/Configuration/AdminConfiguration.php [hunk 8]
```diff
@@ -62,7 +62,8 @@ public function configureOptions(OptionsResolver $resolver)
             'translation_pattern' => $this
                 ->applicationConfiguration
                 ->getParameter('translation')['pattern'],
-            'form' => null
+            'form' => null,
+            'pager' => 'pagerfanta'
```
Comment (johnkrovitch) [inline Admin/Configuration/AdminConfiguration.php]: #82 has been created

File: Filter/PagerfantaFilter.php (hunk 2 above)
Comment (johnkrovitch) [inline Filter/PagerfantaFilter.php]: The adapter is here to provide an interface between the data provider and the pager. It does not hold the request

**Pull request merged**